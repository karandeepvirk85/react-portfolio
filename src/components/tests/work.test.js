import React from 'react';
import Work from "../work.component";
import renderer from 'react-test-renderer'
import {expect,test} from '@jest/globals';

test('Work SnapShot',()=>{
    const componentToTest = renderer.create(<Work/>).toJSON()
    expect(componentToTest).toMatchSnapshot()
})