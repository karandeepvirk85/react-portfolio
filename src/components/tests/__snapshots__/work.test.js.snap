// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Work SnapShot 1`] = `
<div
  className="page-container work"
>
  <h1>
    WORK EXPERIENCE
  </h1>
  <h2>
    React Developer
  </h2>
  <div
    className="job-address"
  >
    <h3>
      Volante Systems, Toronto, ON 
      <strong>
         CANADA 
      </strong>
      <span>
        Feb 2022 - Present
      </span>
    </h3>
    <h4>
      <a
        href="https://www.volantesystems.com/"
        target="_blank"
      >
        https://www.volantesystems.com/
      </a>
    </h4>
  </div>
  <ul>
    <li>
      Developed and tested react components using React, Typescript, ES6, mobX, Formik, Webpack, Jira, Bit bucket, Docker Container, Mongo DB, React testing library, Jest, Sonar cloud, Husky, Mocha and JS hint.
    </li>
    <li>
      Converted designed mock-ups into fully functioning user interface with Unit testing, Integrated Testing under Sprints, Scrums and Agile development environment.
    </li>
    <li>
      Modified discount feature to manage the different types of discounts with the ability to add discounts per item, by percentage or value, ability to set minimum and maximum value, tax options, etc.
    </li>
    <li>
      Developed schedule interface and functionality to modify different types of schedule which can be repeated daily, weekly, monthly, etc.
    </li>
    <li>
      Attended different meetings every day to refine stories, story points, bug triage, weekly refinements. Removed story defects, bugs and worked collaboratively with QA team.
    </li>
    <li>
      Communicated with back end developer and used Swagger to manage end points and handled requests and responses back and forth with server.
    </li>
  </ul>
  <h2>
    React Web Developer
  </h2>
  <div
    className="job-address"
  >
    <h3>
      AlluraDirect, Vancouver BC, 
      <strong>
        CANADA
      </strong>
        
      <span>
        Sep 2020 - Oct 2021
      </span>
    </h3>
    <h4>
      <a
        href="https://www.alluradirect.com/"
        target="_blank"
      >
        https://www.alluradirect.com/
      </a>
    </h4>
  </div>
  <ul>
    <li>
      Developed property manager dashboard interface using Formik, bootstrap calendar, Type Script and ECMAScript (ES6), Jira, Git and Postman to manage the properties, track the issues and to control the versions of the project.
    </li>
    <li>
      Developed react functional components, custom hooks.
    </li>
    <li>
      Managed project states and lifecycle with Redux also prop drilled the states from one component to others.
    </li>
    <li>
      Reviewed and requested code changes to improve the code quality and maintain the code standard before merging the Pull Request to master branch.
    </li>
  </ul>
  <h2>
    Web Developer / Programmer
  </h2>
  <div
    className="job-address"
  >
    <h3>
      Navigator Multimedia, Kelowna BC, 
      <strong>
        CANADA
      </strong>
       
      <span>
        July 2016 - April 2020
      </span>
    </h3>
    <h4>
      <a
        href="https://www.navigator.ca/?"
        target="_blank"
      >
        https://www.navigator.ca/
      </a>
    </h4>
  </div>
  <ul>
    <li>
      Converted designed mock-ups into fully functioning user interface with Unit testing, Integrated Testing under Sprints, Scrums and Agile development environment using DevOps, Jira, Gitlab, GitHub.
    </li>
    <li>
      Integrated various third-party Payments, CRM, Social and Ecommerce REST APIs (Mentioned in Skills).
    </li>
    <li>
      Learned to interact with clients and gained experience, resolved conflicts, and assisted other developers to implement the client’s ideas and increased their revenue.
    </li>
    <li>
      Developed WordPress Themes and Plugins with different user levels, views, and roles to solve complex logics and problems for end users, stakeholders and local businesses working collaboratively and collectively with DRY and MVC coding using Semantic HTML.
    </li>
    <li>
      Enhanced and Integrated Various WordPress plugins (Mentioned in Skills) by adding custom fields, attributes, meta boxes, etc.
    </li>
    <li>
      Data Warehousing, Data normalization, Sorting, Searching, Filters, Autocomplete, Social feeds, Cloud Computing, Pagination, Webhooks, Email Templates, SVG Interactive Map, XML Generation, Data import and export in different file formats like CSV, Excel, Code Minification, dynamic PDF generation using TCPDF, Quizzes in steps for users to choose appropriate option such as events, trails, tours.
    </li>
    <li>
      Ecommerce stores with Reoccurring payment, Partial payment, Refund, Email notifications, Invoice generation, fully functioning shopping cart integrated with Stripe, PayPal Pro.
    </li>
    <li>
      Implemented Google technologies such as Bus transit feed, Markers, Trails, Polylines, Polygon, Waypoints, Clusters, Info Windows, Shapes, KML, Geometry Library, Places Library and Calendar.
    </li>
  </ul>
  <h2>
    Software Developer
  </h2>
  <div
    className="job-address"
  >
    <h3>
      McNewton, IT Park, Chandigarh, 
      <strong>
        INDIA
      </strong>
      <span>
         Dec 2013 - January 2015
      </span>
    </h3>
    <h4>
      <a
        href="https://www.zaubacorp.com/company/McNewton-Solutions-Private-Limited/U72900DL2012PTC234560"
      >
        McNewton (Striked Off)
      </a>
    </h4>
  </div>
  <ul>
    <li>
      Designed and developed e-Learning Web application for students to learn programming using theme and SQL Server.
    </li>
    <li>
      Designed and developed company website and learned basic HTML, CSS, and Some concepts of JavaScript.
    </li>
    <li>
      Assisted engineering students with coding and helped them to pursue their computer science bachelor degree internship.
    </li>
    <li>
      Designed business process documentation, identified various stakeholders for the system, and validated all elements of the requirement specifications document (SRS).
    </li>
    <li>
      Applied multi-paradigm software development methodology such as Unified Modelling Language (UML), Modularity, Reusability, Waterfall model, and Agile development.
    </li>
  </ul>
</div>
`;
